grails-app/controllers/org/pih/warehouse/user/DashboardController.groovy

```diff
@@ -69,14 +69,16 @@ class DashboardController {
             return
         }
 
-        def requisition = Requisition.findByRequestNumber(params.searchTerms)
-        if (requisition) {
+        List<Requisition> requisitions = Requisition.findAllByRequestNumber(params.searchTerms, [sort: 'dateCreated'])
+        if (requisitions) {
```

Comment:
doesn't `findAllBy` return `[]` if there are no records? If so, empty list is truthy, so it will always go inside if and crash when doing `[].first()`
This is why we need to do `requisitions.size()`

Reply:
`findAllBy...()` does return an empty list when it does not match any records
empty list `[]` in groovy evaluates to `false` 
https://docs.groovy-lang.org/latest/html/documentation/#_list_as_a_boolean_expression

Reply:
oh ok, so it's different than in JS, but I remember at some point I got "destroyed" by Groovy when not adding this `.size()`, but I do not remember the exact case now